src/vs/workbench/electron-browser/shell.ts

```diff
@@ -508,13 +512,77 @@ export class WorkbenchShell extends Disposable {
 		// Keep font info for next startup around
 		saveFontInfo(this.storageService);
 
+		this._savePartsSplash();
+
 		// Dispose Workbench
 		if (this.workbench) {
 			this.workbench.dispose(reason);
 		}
 	}
+
+	private _savePartsSplash() {
+
+		// capture html-structure
+		let html = '<div id="monaco-parts-splash">';
+
+		// title part
+		let titleHeight: number;
+		{
+			let part = this.workbench.getContainer(Parts.TITLEBAR_PART);
+			let pos = getDomNodePagePosition(part);
+			let bg = part.style.backgroundColor || 'inhert';
+			html += `<div style="position: absolute; width: 100%; left: 0; top: 0; height: ${pos.height}px; background-color: ${bg};"></div>`;
+			titleHeight = pos.height;
+		}
+
+		// activitybar-part
+		let left = this.workbench.getSideBarPosition() === Position.LEFT;
+		let activityPartWidth: number;
+		{
```

Comment:
All these parts all already aware of their `heigth` so I think no `height` computation is necessery, you can just get their heigth.

Reply:
Same for `width`.

If we remove the `width` and `heigth` computation maybe we could reuse some of this code to not compute all these 4 times

Reply:
@isidorn How/where can I access those dimensions?

Reply:
@jrieken I thought something like 
```javascript
let part = this.workbench.getContainer(Parts.ACTIVITYBAR_PART);
part.clientHeight
```